### PR TITLE
Typescript issue fixes

### PR DESCRIPTION
**Pull request by SiegfriedSobkowiak**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [ ] A peer-review has been executed
  - [ ] The code has been manually inspected by someone who did not implement the feature
  - [ ] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [ ] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [ ] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [ ] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-frontend/src/components/general/DatasetsTabMenu.vue [hunk 1]
```diff
@@ -125,13 +126,13 @@ export default defineComponent({
...
       if (this.initialTabIndex != event.index) {
-        await this.$router.push(this.tabs[event.index].route);
+        await router.push(this.tabs[event.index].route);
       }
     },
   },
 });
 
-interface Tab {
+export interface Tab {
```
Comment (emanueldfine) [inline dataland-frontend/src/components/general/DatasetsTabMenu.vue:135]: Das `export` kann wieder entfernt werden.  Stammt von mir, aber ist nicht nötig.

File: dataland-frontend/src/components/general/DatasetsTabMenu.vue [hunk 2]
```diff
@@ -23,6 +23,7 @@ import TabPanel from 'primevue/tabpanel';
 import { checkIfUserHasRole, KEYCLOAK_ROLE_ADMIN, KEYCLOAK_ROLE_REVIEWER } from '@/utils/KeycloakUtils';
 import type Keycloak from 'keycloak-js';
 import { CompanyRole, type CompanyRoleAssignment } from '@clients/communitymanager';
+import router from '@/router';
```
Comment (emanueldfine) [inline dataland-frontend/src/components/general/DatasetsTabMenu.vue:27]: Der "data" Block ist nicht type-safe (benutzt `as Tab[]`).
Man kann ihn type-safe machen:

`data(): { tabs: Tab[] } {
    return {
    tabs: [
      {
        label: 'COMPANIES',
        route: '/companies',
        isVisible: true,
      },
      {
        label: 'MY DATASETS',
        route: '/datasets',
        isVisible: true,
      },
      {
        label: 'QA',
        route: '/qualityassurance',
        isVisible: false,
      },
      {
        label: 'MY DATA REQUESTS',
        route: '/requests',
        isVisible: true,
      },
      {
        label: 'DATA REQUESTS FOR MY COMPANIES',
        route: '/companyrequests',
        isVisible: false,
      },
      {
        label: 'ALL DATA REQUESTS',
        route: '/requestoverview',
        isVisible: false,
      },
    ],
    }
  },`

File: dataland-frontend/src/components/resources/apiKey/CreateApiKeyCard.vue [hunk 3]
```diff
@@ -149,6 +149,16 @@ export default defineComponent({
     expiryDateFormated(): string {
       return calculateExpiryDateAsDateString(assertDefined(this.expiryTimeDays));
     },
+    customDate: {
+      get() {
+        // Convert number to a Date object
+        return this.customDateInMilliseconds ? new Date(this.customDateInMilliseconds) : null;
+      },
+      set(value: Date | null) {
+        // Convert Date object back to number
```
Comment (emanueldfine) [inline dataland-frontend/src/components/resources/apiKey/CreateApiKeyCard.vue:158]: comments bitte noch löschen

File: dataland-frontend/vite.config.mts [hunk 4]
```diff
@@ -48,6 +48,11 @@ export default defineConfig({
       'axios',
       'i18n-iso-countries',
       'currency-codes/data',
+      'dompurify',
+      'vue-markdown-render',
+      'markdown-it',
+      '@faker-js/faker',
+      'primevue/icons/chevronup',
```
Comment (emanueldfine) [inline dataland-frontend/vite.config.mts:55]: Wurde das hier ergänzt, um die build performance zu verbessern?

File: dataland-frontend/src/components/pages/QualityAssurance.vue [hunk 5]
```diff
@@ -238,15 +248,15 @@ export default defineComponent({
...
+        const selectedFrameworksAsSet = new Set<GetInfoOnUnreviewedDatasetsDataTypesEnum>(
+          this.selectedFrameworks.map((selectableItem) => this.convertDataType(selectableItem.frameworkDataType))
         );
         const reportingPeriodFilter: Set<string> = new Set<string>(
           this.availableReportingPeriods?.map((date) => date.getFullYear().toString())
         );
         const companyNameFilter = this.searchBarInput === '' ? undefined : this.searchBarInput;
         const response = await this.apiClientProvider.apiClients.qaController.getInfoOnUnreviewedDatasets(
-          selectedFrameworksAsSet as Set<GetInfoOnUnreviewedDatasetsDataTypeEnum>,
+          selectedFrameworksAsSet as Set<GetInfoOnUnreviewedDatasetsDataTypesEnum>,
```
Comment (emanueldfine) [inline dataland-frontend/src/components/pages/QualityAssurance.vue:259]: Das `as` kann weg, weil du jetzt type-safe sein solltest und den compiler nicht mehr cheaten musst.

File: dataland-frontend/src/components/pages/QualityAssurance.vue [hunk 6]
```diff
@@ -255,7 +265,7 @@ export default defineComponent({
         this.displayDataOfPage = response.data;
         this.totalRecords = (
           await this.apiClientProvider.apiClients.qaController.getNumberOfUnreviewedDatasets(
-            selectedFrameworksAsSet as Set<GetInfoOnUnreviewedDatasetsDataTypeEnum>,
+            selectedFrameworksAsSet as Set<GetInfoOnUnreviewedDatasetsDataTypesEnum>,
```
Comment (emanueldfine) [inline dataland-frontend/src/components/pages/QualityAssurance.vue:268]: Das `as`  kann weg, weil du jetzt type-safe sein solltest und den compiler nicht mehr cheaten musst.

File: dataland-frontend/src/components/resources/apiKey/CreateApiKeyCard.vue (hunk 3 above)
Comment (jenniferlechner) [reply dataland-frontend/src/components/resources/apiKey/CreateApiKeyCard.vue:158]: done

File: dataland-frontend/vite.config.mts (hunk 4 above)
Comment (jenniferlechner) [reply dataland-frontend/vite.config.mts:55]: removed them

File: dataland-frontend/src/components/pages/QualityAssurance.vue [hunk 7]
```diff
@@ -230,6 +232,14 @@ export default defineComponent({
   methods: {
     convertUnixTimeInMsToDateString,
     humanizeString: humanizeStringOrNumber,
+    /**
+     * Converts from DataTypeEnum to GetInfoOnUnreviewedDatasetsDataTypesEnum
+     * @param input DataTypeEnum
+     * @returns GetInfoOnUnreviewedDatasetsDataTypesEnum
+     */
+    convertDataType(input: DataTypeEnum): GetInfoOnUnreviewedDatasetsDataTypesEnum {
```
Comment (emanueldfine) [inline dataland-frontend/src/components/pages/QualityAssurance.vue:240]: Ich schlage vor, die Funktion nicht `convert...` zu nennen, da du hier nicht konvertierst.
Du cheatest theoretisch nur, indem du dem compiler sagst 
"behandle `input` ab jetzt bitte so, als sei es vom Typ `GetInfoOnUnreviewedDatasetsDataTypesEnum`, ganz gleich ob es passt oder nicht".

Daher wäre es vermutlich besser, sie `manuallyChangeTypeOfDataTypeEnum` oder sowas.

Ein "echter" convert würde übrigens in etwa so aussehen:

```
function convertToGetInfoEnum(dataType: DataTypeEnum): GetInfoOnUnreviewedDatasetsDataTypesEnum {
    if (Object.values(GetInfoOnUnreviewedDatasetsDataTypesEnum).includes(dataType as GetInfoOnUnreviewedDatasetsDataTypesEnum)) {
        return dataType as GetInfoOnUnreviewedDatasetsDataTypesEnum;
    }
    throw new Error(`Invalid DataTypeEnum value: ${dataType}`);
}

```

File: dataland-frontend/vite.config.mts (hunk 4 above)
Comment (emanueldfine) [reply dataland-frontend/vite.config.mts]: Wieso.  Kommt scheinbar von Siegfried.  Macht vllt Sinn.

@SiegfriedSobkowiak

Comment (emanueldfine) [reply dataland-frontend/vite.config.mts]: @jenniferlechner  habe mit Siegfried gesprochen.
Der preload der Bibliotheken hier beschleunig den build von vite.
Daher bitte wieder einfügen.

**Pull request merged**